core/src/main/java/org/frankframework/management/bus/endpoints/FileViewer.java

```diff
@@ -0,0 +1,92 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.frankframework.management.bus.endpoints;
+
+
+import org.apache.commons.io.FilenameUtils;
+import org.frankframework.management.bus.ActionSelector;
+import org.frankframework.management.bus.BinaryResponseMessage;
+import org.frankframework.management.bus.BusAction;
+import org.frankframework.management.bus.BusAware;
+import org.frankframework.management.bus.BusException;
+import org.frankframework.management.bus.BusMessageUtils;
+import org.frankframework.management.bus.BusTopic;
+import org.frankframework.management.bus.ResponseMessageBase;
+import org.frankframework.management.bus.TopicSelector;
+import org.frankframework.util.AppConstants;
+import org.frankframework.util.FileUtils;
+import org.springframework.http.MediaType;
+import org.springframework.messaging.Message;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+@BusAware("frank-management-bus")
+@TopicSelector(BusTopic.FILE_VIEWER)
+public class FileViewer extends BusEndpointBase {
+
+	public static final String permissionRules = AppConstants.getInstance().getProperty("FileViewer.permission.rules");
+
+	@ActionSelector(BusAction.GET)
+	public Message<?> getFileContent(Message<?> message) {
+		String resultType = BusMessageUtils.getHeader(message, "resultType");
+		String fileName = BusMessageUtils.getHeader(message, "fileName");
+		ResponseMessageBase<?> response;
+
+		if (fileName == null || resultType == null) {
+			throw new BusException("fileName or type not specified");
+		} else if (!FileUtils.readAllowed(permissionRules, fileName, BusMessageUtils::hasRole)){
+			throw new BusException("not allowed");
```

Comment:
Misschien leuk om hier de recent gemaakte `BusException("not allowed", 403)` terug te geven.

Reply:
Ah het zit nog niet in deze branch dus moet master hierin mergen